initial-reverse/reverse.cc

```diff
@@ -0,0 +1,126 @@
+/*
+ * =====================================================================================
+ *
+ *       Filename:  reverse.cc
+ *
+ *    Description:  Reverses the lines of a file
+ *
+ *        Version:  1.0
+ *        Created:  07/04/2020 14:49:10
+ *       Revision:  none
+ *       Compiler:  g++
+ *
+ *         Author:  Brian Li,
+ *   Organization:
+ *
+ * =====================================================================================
+ */
+#include <iostream>
+#include <stdexcept>
+#include <sys/stat.h>
+
+template <class T> class Vector {
+  // Backing array
+  T *arr;
+  int cap;
+  int len;
+
+public:
+  // Constructor
+  Vector() {
+    cap = 1;
+    len = 0;
+
+    // Initialize a to point to an array of length `cap`
+    arr = (T *)malloc(cap * sizeof(T));
+  }
+
+  // Inserts `v` at the end
+  void insert(const T &v) {
+    if (len == cap) {
+      // Copy `arr` into a new array
+      cap *= 2;
+      T *nextArr = (T *)malloc(cap * sizeof(T));
+      for (int i = 0; i < len; i++) {
+        nextArr[i] = arr[i];
+      }
+      arr = nextArr;
+    }
+
+    arr[len] = v;
+    len += 1;
+  }
+
+  // Gets the last element
+  T get() { return arr[len - 1]; }
```

Comment:
do you want it to return a const reference instead? can the member function be marked const too?